### PR TITLE
Add cryptoauthlib with EEPROM support for certificates

### PR DESCRIPTION
**Pull request by GalaxyGorilla**

### CODE REVIEW

File: src/grisp_hw.erl [hunk 1]
```diff
@@ -1,20 +1,97 @@
 % @doc GRiSP hardware access.
 -module(grisp_hw).
 
+-include("grisp_hw.hrl").
+
 % API
 -export([platform/0]).
+-export([read_eeprom/0]).
```
Comment (eproxus) [inline src/grisp_hw.erl:8]: I think we should prefix the EEPROM functions with `eeprom_` instead, so they're logically grouped.

File: src/grisp_hw.hrl [hunk 2]
```diff
@@ -0,0 +1,41 @@
+-define(GRISP_EEPROM_ADR, 16#57).
+-define(GRISP_EEPROM_DATA_SIZE, 28).
+-define(GRISP_EEPROM_BUS, i2c0).
+
+-record(grisp_hw_state, {eeprom_bus}).
```
Comment (eproxus) [inline src/grisp_hw.hrl:5]: Not used?

File: src/grisp_hw.erl [hunk 3]
```diff
@@ -1,20 +1,97 @@
...
+      VersMajor:1/binary,   %% contains GRiSP version and PCB Major
+      VersMinor:8,          %% actually the PCB minor
+      Variant:8,
+      _Mac:6/binary,
+      _Dummy2:1/binary,     %% unused
+      Crc:2/little-unit:8,
+      _Dummy3:2/binary>> = Data = grisp_ni2c:read(Bus, ?GRISP_EEPROM_ADR,
+                                                  0, ?GRISP_EEPROM_DATA_SIZE),
+    <<GrispVersion:4, PcbMajor:4>> = VersMajor,
+    MetaData = [{grisp_version, lists:flatten(io_lib:format("~p", [GrispVersion]))},
```
Comment (eproxus) [inline src/grisp_hw.erl:58]: Can we use a map here instead?

File: src/grisp_hw.erl [hunk 4]
```diff
@@ -1,20 +1,97 @@
...
+
+% @private
+crc16(Cp) ->
+    crc16(0, Cp).
+
+% @private
+crc16(Crc, <<>>) ->
+    Crc;
+crc16(Crc, <<Cp:8, CpTail/binary>>) ->
+    crc16((Crc bsr 8) bxor lists:nth(((Crc bxor Cp) band 16#ff) + 1, ?misc_crc16_tab), CpTail).
```
Comment (eproxus) [inline src/grisp_hw.erl:97]: I think a tuple accessed with `element/2` would be much faster here.

**Review by eproxus — CHANGES_REQUESTED** [review]

File: src/grisp_hw.erl (hunk 1 above)
Comment (GalaxyGorilla) [reply src/grisp_hw.erl:8]: 👍

File: src/grisp_hw.hrl (hunk 2 above)
Comment (GalaxyGorilla) [reply src/grisp_hw.hrl:5]: ups ....

File: src/grisp_hw.erl (hunk 3 above)
Comment (GalaxyGorilla) [reply src/grisp_hw.erl:58]: hmm ok 👍

File: src/grisp_hw.erl (hunk 4 above)
Comment (GalaxyGorilla) [reply src/grisp_hw.erl:97]: I can change that even though speed here has no real relevance.

**Pull request closed without merging**